### PR TITLE
[ONNX FE] Fix issues of com.microsoft.EmbedLayerNormalization

### PR DESCRIPTION
**Pull request by a3213105**
I have a customized ONNX model which used "EmbedLayerNormalization" op, the struct is shown in picture below. and I have met an error when I load model with OpenVINO. it shows the mismatch of the shape between two input of "ADD" op after "EmbedLayerNormalization".
I have check the original implement of EmbedLayerNormalization in [ONNX-Runtime](https://github.com/microsoft/onnxruntime/blob/0c6037b5abe571fc43a55ef7a9d2f846820fbe5d/onnxruntime/contrib_ops/cpu/bert/embed_layer_norm.cc) , and i found some different between ONNX-Runtime and OpenVINO.
So, I create this pull request to fix this gap between ONNX-Runtime and OpenVINO
1. Add implement of EmbedLayerNormalization which seq_len != pos_embed_len
2. Align EmbedLayerNormalization implement in OpenVINO with ONNX-Runtime
 
### Tickets:
 - #13035
image

### CODE REVIEW

File: src/frontends/onnx/frontend/src/op/com.microsoft/embed_layer_normalization.cpp [hunk 1]
```diff
@@ -27,10 +27,31 @@ OutputVector embed_layer_normalization(const Node& node) {
...
+    // add position embeddings
+    if (num_nodes > 8 && !ngraph::op::is_null(nodes[8])) {
+        // if we have position_ids
+        const auto& position_ids = nodes[8];
+        const auto gathered_position_embeddings =
+            std::make_shared<default_opset::Gather>(position_embeddings, position_ids, zero, 0);
+        input = std::make_shared<default_opset::Add>(input, gathered_position_embeddings);
+    } else {
+        // input_ids's shape is [batchsize, sequence_length]
+        // input'shape is [batchszie, sequence_length, hidden_size]
```
Comment (mateusztabaka) [inline src/frontends/onnx/frontend/src/op/com.microsoft/embed_layer_normalization.cpp:41]:
```suggestion
        // input's shape is [batchsize, sequence_length, hidden_size]
```

File: src/frontends/onnx/frontend/src/op/com.microsoft/embed_layer_normalization.cpp [hunk 2]
```diff
@@ -27,10 +27,31 @@ OutputVector embed_layer_normalization(const Node& node) {
...
+        // position_embeddings's shape is [max_sequence_lenght, hidden_size]
```
Comment (mateusztabaka) [inline src/frontends/onnx/frontend/src/op/com.microsoft/embed_layer_normalization.cpp:42]:
```suggestion
        // position_embeddings's shape is [max_sequence_length, hidden_size]
```

File: src/frontends/onnx/frontend/src/op/com.microsoft/embed_layer_normalization.cpp [hunk 3]
```diff
@@ -27,10 +27,31 @@ OutputVector embed_layer_normalization(const Node& node) {
...
-
+    // add position embeddings
+    if (num_nodes > 8 && !ngraph::op::is_null(nodes[8])) {
+        // if we have position_ids
+        const auto& position_ids = nodes[8];
+        const auto gathered_position_embeddings =
+            std::make_shared<default_opset::Gather>(position_embeddings, position_ids, zero, 0);
+        input = std::make_shared<default_opset::Add>(input, gathered_position_embeddings);
+    } else {
+        // input_ids's shape is [batchsize, sequence_length]
```
Comment (mateusztabaka) [inline src/frontends/onnx/frontend/src/op/com.microsoft/embed_layer_normalization.cpp:40]:
```suggestion
        // input_ids' shape is [batchsize, sequence_length]
```

File: src/frontends/onnx/tests/onnx_import_com_microsoft.in.cpp [hunk 4]
```diff
@@ -405,6 +405,105 @@ NGRAPH_TEST(${BACKEND_NAME}, onnx_model_embed_layer_normalization_dynamic_shapes
...
+        6,
+        5,
+        5,
+    };
+
+    auto test_case = test::TestCase(function, s_device);
+    test_case.add_input<int>(Shape{3, 8}, input_ids);
+    test_case.add_input<int>(Shape{3, 8}, segment_ids);
+    test_case.add_input<float>(Shape{10, 5}, word_embeddings);
+    test_case.add_input<float>(Shape{10, 5}, position_embeddings);
```
Comment (mateusztabaka) [inline src/frontends/onnx/tests/onnx_import_com_microsoft.in.cpp:497]: It'd be good to have a test where position embeddings have different shape than word embeddings.

File: src/frontends/onnx/frontend/src/op/com.microsoft/embed_layer_normalization.cpp [hunk 5]
```diff
@@ -27,10 +27,31 @@ OutputVector embed_layer_normalization(const Node& node) {
...
+        // input'shape is [batchszie, sequence_length, hidden_size]
+        // position_embeddings's shape is [max_sequence_lenght, hidden_size]
+        // output = input + position_embeddings,
+        // after broadcast the shape of output is [batchszie, max_sequence_lenght, hidden_size]
+        // the output shape is error.
```
Comment (mateusztabaka) [inline src/frontends/onnx/frontend/src/op/com.microsoft/embed_layer_normalization.cpp:45]: I'd rephrase that to:
```
therefore input and position_embeddings cannot be added together
so we need slice the position_embeddings ........
```

**Review by mateusztabaka — CHANGES_REQUESTED** [review]
Thanks for contribution. This new version of EmbedLayerNormalization was introduced after we implemented that op in OpenVino.
I have some minor comments that can be addressed.

> Align EmbedLayerNormalization implement in OpenVINO with ONNX-Runtime
Latest version of EmbedLayerNormalization also specifies third optional output 'embedding_sum'. Do you plan to add that as well?

Can you also add some tests where position_ids input is present in the model?

File: src/frontends/onnx/tests/onnx_import_com_microsoft.in.cpp (hunk 4 above)
Comment (a3213105) [reply src/frontends/onnx/tests/onnx_import_com_microsoft.in.cpp]: thanks for your comment, i have updated the test case

File: src/frontends/onnx/frontend/src/op/com.microsoft/embed_layer_normalization.cpp (hunk 5 above)
Comment (a3213105) [reply src/frontends/onnx/frontend/src/op/com.microsoft/embed_layer_normalization.cpp]: update in new commit

**Comment by a3213105** [comment]
> > Align EmbedLayerNormalization implement in OpenVINO with ONNX-Runtime
> > Latest version of EmbedLayerNormalization also specifies third optional output 'embedding_sum'. Do you plan to add that as well?

Sorry, I missed this comment. I will try to implement the  latest version of EmbedLayerNormalization into OpenVINO, but now i don't have sample data for this version. maybe i can try it later.

**Comment by a3213105** [comment]
> Thanks for contribution. This new version of EmbedLayerNormalization was introduced after we implemented that op in OpenVino. I have some minor comments that can be addressed.
> 
> > Align EmbedLayerNormalization implement in OpenVINO with ONNX-Runtime
> > Latest version of EmbedLayerNormalization also specifies third optional output 'embedding_sum'. Do you plan to add that as well?
> 
> Can you also add some tests where position_ids input is present in the model?

I have check the third output, we'd have to unroll MVN to have it, and I don't have time to support it at this time

File: src/frontends/onnx/frontend/src/op/com.microsoft/embed_layer_normalization.cpp [hunk 6]
```diff
@@ -5,6 +5,7 @@
 #include "op/com.microsoft/embed_layer_normalization.hpp"
 
 #include "default_opset.hpp"
+#include "ngraph/log.hpp"
```
Comment (mateusztabaka) [inline src/frontends/onnx/frontend/src/op/com.microsoft/embed_layer_normalization.cpp:8]: this is probably not required

**Review by mateusztabaka — APPROVED** [review]

**Review by praasz — APPROVED** [review]
